### PR TITLE
Cli single upgrade

### PR DESCRIPTION
**Pull request by jdalsem**

### CODE REVIEW

File: engine/tests/phpunit/integration/Elgg/Cli/UpgradeBatchCommandTest.php [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+		$application->add(new UpgradeBatchCommand());
+
+		$command = $application->find('upgrade:batch');
+		$commandTester = new CommandTester($command);
+		$commandTester->execute([
+			'command' => $command->getName(),
+			'upgrades' => ['RandomNonExistingClass'],
+		]);
+
+		$this->assertRegExp('/' . elgg_echo('cli:upgrade:batch:finished') . '/im', $commandTester->getDisplay());
```
Comment (jeabakker) [inline engine/tests/phpunit/integration/Elgg/Cli/UpgradeBatchCommandTest.php]: you can also use `->assertStringContains()`

File: engine/classes/Elgg/Cli/UpgradeBatchCommand.php [hunk 2]
```diff
@@ -0,0 +1,55 @@
+<?php
+
+namespace Elgg\Cli;
+
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Input\InputArgument;
```
Comment (jeabakker) [inline engine/classes/Elgg/Cli/UpgradeBatchCommand.php]: left over junk

File: engine/tests/phpunit/integration/Elgg/Upgrade/LocatorIntegrationTest.php [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+	}
+
+	public function countItems() {
+		return 100;
+	}
+
+	public function run(Result $result, $offset) {
+		$result->addSuccesses(10);
+	}
+}
```
Comment (jeabakker) [inline engine/tests/phpunit/integration/Elgg/Upgrade/LocatorIntegrationTest.php]: aaaarrrrhhhh, luck for you the lint tests don't work on tests

**Review by jeabakker — CHANGES_REQUESTED** [review]

**Comment by jdalsem** [comment]
do not merge yet... additional feat coming

**Review by jeabakker — APPROVED** [review]

**Pull request merged**